File: leetcode3/김민제/3950_Exactly_One_Consecutive_Set_Bits_Pair.py
class Solution:
    def consecutiveSetBits(self, n: int) -> bool:
        count = 0
        binary = bin(n)[2:]

        for i in range(len(binary)-1):
            if binary[i:i+2] == "11":
                count += 1
        return count == 1     

File: leetcode3/염혜정/3950. Exactly One Consecutive Set Bits Pair.java
// 2진수를 구하고 오직 한 쌍의 11만 존재하여야 함
// O(logn)

class Solution {
    public boolean consecutiveSetBits(int n) {
        int cnt = 0;
        int pre = 0;
        while (n>0) {
            if (pre == 1 && n%2 ==1) cnt++;
            pre = n%2;
            n /= 2;
        }
        
        return cnt == 1;
    }
}
